### DISCUSSION TITLE
Discussion: `openmc.deplete` support

### DISCUSSION THREAD

**Discussion by yardasol** [arfc/saltproc#68] [Ideas]
The `openmc.deplete` module enables depletion simulations in OpenMC.

This issue can be closed when functions exists that utilize the `openmc.deplete` module and enable similar functionality  as the following:

- [ ] `Depcode`
- - [ ] `run_depcode`
- - [ ] `replace_burnup_parameters`
- [ ] `Simulation`
- - [ ] `runsim_no_reproc`
- [ ] `App`
- - [ ] `run`

**Comment by yardasol** [comment]
# `Depcode` Discussion
## `run_depcode`
This function sends instructions and flags to the depletion code executable. OpenMC can also be run this way. I can think of two ways to support OpenMC:
1. Create an entirely new function that utilizes the OpenMC API. This may require renaming `run_depcode` to `run_serpent`
```python
def run_serpent(self, cores, nodes):
    """ Runs serpent code as subprocess with the given parameters.

    Parameters
    ----------
    .
    .
    .
    """
    .
    .
    .

def run_openmc(self, cores, nodes):
    """ Runs OpenMC code as subprocess with the given parameters.

    Parameters
    ----------
    .
    .
    .
    """
    .
    .
    .

```
2. Add an `if`-statement to the function to specify which depletion code we are using based on the `codename` field of the `Depcode` class ('serpent' for Serpent, 'openmc' for OpenMC), and then refactor the current function into the body of one of the `if`-statements OR a collection of helper functions for each depletion code option. Below I have shown the former.
```python
def run_depcode(self, cores, nodes):
    """ Run depletion code as subprocess with the given parameters.

    Parameters
    ----------
    .
    .
    .
    """
    if (self.codename == "serpent"):
        # repeat body of current run_depcode here
        .
        .
        .
    else if (self.codename == "openmc"):
        # Use OpenMC CLI's here
        .
        .
        .
```
Given that `Depcode` already has the `codename` field built-in, I think approach 2 makes more sense for this function.

## `replace_burnup_parameters`

**Comment by yardasol** [comment]
> ## `Depcode.run_depcode`
> 
> This function sends instructions and flags to the depletion code executable. OpenMC can also be run this way.

While OpenMC itself can be run from the terminal, it seems that the `deplete` module is has been written to work exclusively from the Python API based on [this](https://openmc.discourse.group/t/openmc-depletion-via-terminal/1256) post in the OpenMC discourse group. @gwenchee seems to have come up with a workaround to this by putting all of the necessary object instantiations a script which we can see in her post [here](https://openmc.discourse.group/t/depletion-simulation-on-supercomputer-gets-stuck-between-depletion-steps/606). Adding a function that would construct such a script utilizing the currently proposed functions in #55, as well as relevant methods from the `openmc.deplete` package, seems to be the best way forward right now.

**Comment by gwenchee** [comment]
This makes sense. The OpenMC team might have added more functionality for this in the past year, you should take a closer look at that.
